server/rest/src/test/java/org/infinispan/rest/Http2Test.java

```diff
@@ -47,14 +49,22 @@ public void afterMethod() {
       Util.close(client);
    }
 
-   @Test
-   public void shouldUseHTTP1WithALPN() {
-      secureUpgradeTest(HTTP_11);
+   @DataProvider(name = "openssl-item")
+   public Object[][] opensslItemProvider() {
+      return new Object[][] {
+            {SslContextFactory.SslContextName.DEFAULT.getName()},
```

Comment:
Instead of hardcoding the providers, this method should build the list dynamically based on the availability of openssl. The default provider is obviously always available.

Reply:
The point is that `openssl` is mandatory for RHEL.
Should I create a test argument that: if openssl is not present in the host machine ( the machine that runs the test suite ) return a failure instead of not running the test?

Reply:
Also, based on the availability of `openssl` will require that `commons-test` has a dependency from `commons/all`.
I decided to go with `SslContextName.PROVIDER`.